test/gerkin/features/steps/data_list.js

```diff
@@ -0,0 +1,97 @@
+const { spec, expect, stores } = require('pactum');
+const { When, Then, Given, After } = require('@cucumber/cucumber');
+
+const firstSearchedName = 'John';
+const secondSearchedName = 'Adrien';
+const baseUrl = 'http://localhost:3333/data/registry1/version1';
+const header = {
+  key: 'Information-Mediator-Client',
+  value: 'INSTANCE/CLASS/MEMBER/SUBSYSTEM',
+};
+
+// scenario 1
+Given('The user wants to search for "John Helmut" in the database', () => {
+  return firstSearchedName;
+});
+
+Given('The searched value exists in several records in the database', () => {
+  return 'Searched value does exist in the database';
```

Comment:
Is there a way we could actually check if the user exists in the database but without using the API we are testing?
For our mocked example application we know it exists but what about other future implementations? How can we ensure/check that this test user exists? Maybe we should use more abstract name instead of 'John Helmut', for example 'Test User'

Reply:
This is an excellent question, from perspective of the test harness I think only way to do the check is actual API call. 

Test harness could be configured before scenarios are executed. We might need to introduce it with more advanced test scenarios in the future. Quick idea for this would be having Config.js interface that'd provide setup methods called accordingly in Cucumber's `Before` hook. During test execution config would be replaced with one developed for `/example/<application>` purpose.

In our case I don't think we need to introduce any kind of additional config. We expect `John Helmut` to exist and it should be test instance responsibility to provide it (could be done during execution of `test_entrypoint.sh`, for example as part of Dockerfile). 

Another interesting point is what type of naming convention we should take. `John Helmut` is taken directly from openAPI schema definition, therefore we used it in here. On one hand abstract names can be more descriptive, but on the other it is synthetic and maybe on behavior tests level we might want to keep it more story-like. Personally I'd prefer to keep fake names instead of test strings but it's something we should discuss with community.